e2e/playwright/text-to-cad-tests.spec.ts

```diff
@@ -689,3 +691,749 @@ test(
+      await expect(page.getByTestId('app-header-project-name')).toContainText(
+        projectName
+      )
+      await expect(page.getByTestId('app-header-file-name')).toBeVisible()
+      await expect(page.getByTestId('app-header-file-name')).toContainText(
+        '2x2x2-cube.kcl'
+      )
+
+      // Check file is created
+      await u.openFilePanel()
+      await expect(
+        page.getByTestId('file-tree-item').getByText('2x2x2-cube.kcl')
+      ).toBeVisible()
+    }
+  )
+
+  test(
+    'Home Page -> Text To CAD -> New Project -> Navigate to different project -> Reject -> should stay in project',
+    { tag: '@electron' },
+    async ({ context, page, homePage }, testInfo) => {
+      const u = await getUtils(page)
+      const projectName = 'my-project-name'
+      const unrelatedProjectName = 'unrelated-project'
+      const prompt = '2x2x2 cube'
+      await mockPageTextToCAD(page)
+
+      // Create and navigate to the project then come home
+      await createProject({
+        name: unrelatedProjectName,
+        page,
+        returnHome: true,
+      })
+
+      await expect(page.getByText('Your Projects')).toBeVisible()
+
+      // open commands
+      await page.getByTestId('command-bar-open-button').click()
+
+      // search Text To CAD
+      await page.keyboard.type('Text To CAD')
+      await page.keyboard.press('Enter')
+
+      // new project
+      await page.keyboard.type('New project')
+      await page.keyboard.press('Enter')
+
+      // write name
+      await page.keyboard.type(projectName)
+      await page.keyboard.press('Enter')
+
+      // prompt
+      await page.keyboard.type(prompt)
+      await page.keyboard.press('Enter')
+
+      await homePage.openProject(unrelatedProjectName)
+      // Check that we opened the unrelated project
+      await expect(page.getByTestId('app-header-project-name')).toBeVisible()
+      await expect(page.getByTestId('app-header-project-name')).toContainText(
+        unrelatedProjectName
+      )
+      await expect(page.getByTestId('app-header-file-name')).toBeVisible()
+      await expect(page.getByTestId('app-header-file-name')).toContainText(
+        'main.kcl'
+      )
+
+      await page.getByRole('button', { name: 'Reject' }).click()
+
+      // Check header is populated with the project and file name
+      await expect(page.getByTestId('app-header-project-name')).toBeVisible()
+      await expect(page.getByTestId('app-header-project-name')).toContainText(
+        unrelatedProjectName
+      )
+      await expect(page.getByTestId('app-header-file-name')).toBeVisible()
+      await expect(page.getByTestId('app-header-file-name')).toContainText(
+        'main.kcl'
+      )
+
+      // Check file is created
+      await u.openFilePanel()
+      // File should be deleted
+      await expect(
+        page.getByTestId('file-tree-item').getByText('2x2x2-cube.kcl')
+      ).not.toBeVisible()
+
+      await u.goToHomePageFromModeling()
```

Comment:
In future, you should get rid of the standalone util function in favor of the `toolbar.logoLink` fixture locator like this example here https://github.com/KittyCAD/modeling-app/blob/main/e2e/playwright/projects.spec.ts#L1098-L1098